larpandoracontent/LArTwoDReco/HitWidthClusterMergingAlgorithm.cc

```diff
@@ -0,0 +1,503 @@
+                  continue;
+
+              const ClusterAssociationMap::const_iterator secondaryMapIter = clusterAssociationMap.find(pPrimaryCluster);
+
+	          // if primary cluster has no associations (this shouldn't ever be the case)
+              if(secondaryMapIter == clusterAssociationMap.end()) 
+                  continue;
+
+              const ClusterSet &secondaryForwardAssociations(secondaryMapIter->second.m_forwardAssociations);
+
+              // if cluster is present in the forward associations of any cluster at the same level remove
+              if(secondaryForwardAssociations.find(pConsideredCluster) != secondaryForwardAssociations.end()) 
+              {
+                  ClusterSet &tempPrimaryForwardAssociations(tempMap.find(pCluster)->second.m_forwardAssociations);
+                  const ClusterSet::const_iterator forwardAssociationToRemove(tempPrimaryForwardAssociations.find(pConsideredCluster));
+
+                  // if association has already been removed
+                  if(forwardAssociationToRemove == tempPrimaryForwardAssociations.end())
+                      continue;
+
+                  ClusterSet &tempPrimaryBackwardAssociations(tempMap.find(pConsideredCluster)->second.m_backwardAssociations);
+                  const ClusterSet::const_iterator backwardAssociationToRemove(tempPrimaryBackwardAssociations.find(pCluster));
+
+                  // if association has already been removed
+                  if(backwardAssociationToRemove == tempPrimaryBackwardAssociations.end())
+                      continue;
+
+                  tempPrimaryForwardAssociations.erase(forwardAssociationToRemove);
+                  tempPrimaryBackwardAssociations.erase(backwardAssociationToRemove);
+              }
+          }
+      }
+  }
+  clusterAssociationMap = tempMap;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+bool HitWidthClusterMergingAlgorithm::AreClustersAssociated(const LArHitWidthHelper::ClusterParameters &currentFitParameters, const LArHitWidthHelper::ClusterParameters &testFitParameters) const
+{
+
+    CartesianVector currentClusterDirection(0,0,0);
+    CartesianVector testClusterDirection(0,0,0);
+
+    if(m_useSlidingLinearFit)
+    {
+        const CartesianPointVector currentConstituentHitPositionVector(LArHitWidthHelper::GetConstituentHitPositionVector(currentFitParameters.GetConstituentHitVector()));
+        const CartesianPointVector testConstituentHitPositionVector(LArHitWidthHelper::GetConstituentHitPositionVector(testFitParameters.GetConstituentHitVector()));
+
+        const float layerPitch(LArGeometryHelper::GetWireZPitch(this->GetPandora()));
+
+        TwoDSlidingFitResult currentFitResult(&currentConstituentHitPositionVector, m_layerFitHalfWindow, layerPitch);
+        TwoDSlidingFitResult testFitResult(&testConstituentHitPositionVector, m_layerFitHalfWindow, layerPitch);
```

Comment:
const variables whenever possible, including TwoDSlidingFitResults and StatusCodes